platforms/chibios/drivers/audio_dac_basic.c

```diff
@@ -48,6 +48,15 @@
 #    define AUDIO_PIN_ALT -1
 #endif
 
+// Check to see if the DAC triggers are set. If not set them to the defaults for the STM32F303
+#if !defined(AUDIO_DAC_CH1_TRIGGER)
+#    define AUDIO_DAC_CH1_TRIGGER 0b000
+#endif
+
+#if !defined(AUDIO_DAC_CH2_TRIGGER)
+#    define AUDIO_DAC_CH2_TRIGGER 0b010
+#endif
```

Comment:
One possible option is to do something like this:
```suggestion
// The DAC trigger values are different between the STM32 MCU families.
#if !defined(AUDIO_DAC_CH1_TRIGGER)
#    if defined(QMK_MCU_SERIES_STM32G4XX)
#        define AUDIO_DAC_CH1_TRIGGER 0b0111
#    else
#        define AUDIO_DAC_CH1_TRIGGER 0b000
#    endif
#endif

#if !defined(AUDIO_DAC_CH2_TRIGGER)
#    if defined(QMK_MCU_SERIES_STM32G4XX)
#        define AUDIO_DAC_CH2_TRIGGER 0b0010
#    else
#        define AUDIO_DAC_CH2_TRIGGER 0b010
#    endif
#endif
```
(The value for `TIM7_TRGO` happens to be the same for these families.)

Maybe we should even list the supported families explicitly (this may be tricky though, because there are also STM32 lookalikes from other manufacturers which might work with this driver).

Finally, this code may be moved to `audio_dac.h`, so that it could be shared between the additive and basic variants.

Reply:
```suggestion
// The DAC trigger values are different between the STM32 MCU families.
	#if !defined(AUDIO_DAC_CH1_TRIGGER)
	#    if defined(QMK_MCU_SERIES_STM32G4XX)
	#        define AUDIO_DAC_CH1_TRIGGER 0b0111
	#    else
	#        define AUDIO_DAC_CH1_TRIGGER 0b000
	#        if !defined(QMK_MCU_SERIES_STM32F3)
	#            #warning "Default trigger value assigned for AUDIO_DAC_CH1_TRIGGER please ensure this is correct for your MCU"
	#        endif
	#    endif
	#endif
	
	#if !defined(AUDIO_DAC_CH2_TRIGGER)
	#    if defined(QMK_MCU_SERIES_STM32G4XX)
	#        define AUDIO_DAC_CH2_TRIGGER 0b0010
	#    else
	#        define AUDIO_DAC_CH2_TRIGGER 0b010
	#        if !defined(QMK_MCU_SERIES_STM32F3)
	#            #warning "Default trigger value assigned for AUDIO_DAC_CH2_TRIGGER please ensure this is correct for your MCU"
	#        endif
	#    endif
	#endif
```

Would something like work? The define gating the warning can be extended to include all families currently supported, and yes that is me offering to check the data sheets. Then as new families are added there will be a clear warning to make users check the allocation?

I'm happy to move the above code, and the long comment about the trigger mapping to `audio_dac.h`. However only one conversion configuration is used in `audio_dac_additive.c` so there will be a redundant define, is that okay?

Reply:
`#warning` would be equivalent to `#error`, because QMK uses `-Werror` by default; some code uses `#pragma message "foo"` instead, but usually nobody looks at those messages.

Also apparently the same trigger values are used on F0xx, F1xx, F4xx chips, maybe even more; L4xx might be annoying due to having a “L4xx+” family, but apparently those “plus” chips are not yet supported.

As for a redundant define, it probably won't matter if you won't have `#error` there, and maybe the additive driver eventually could be enhanced to use the second channel too (e.g., to output the opposite phase signal).

Reply:
I've implemented the code with the pragma message. After a fair bit of time trying to debug why the DAC just wasn't working i feel leaving a hint for whoever wants to add support for the `L4XX+` or another MCU is  a helpful thing. Most people shouldn't ever see the message.

I also went through all the STM32 familes i could see `QMK_MCU_SERIES_STM32XXX` defines for and checked thier trigger tables. They all match the `STM32F303` except the `STM32F4` which has a very limited set of triggers

For anyone in the future the `STM32L4XX+` has a differnet set of mappings again, timer 6 is `0b0101` timer 7 is `0b0110`

Reply:
I checked the `STM32F410` and it only supported Timer 5. I missed that the `STM32F405` and `STM32F407` have the other timers. I've added them in.